src/main/kotlin/racingcar/domain/numbergenerator/StubNumberGenerator.kt

```diff
@@ -0,0 +1,7 @@
+package racingcar.domain.numbergenerator
+
+class StubNumberGenerator(
+    private val fakeNumbers: MutableList<Int>,
```

Comment:
```suggestion
    private val fakeNumbers: List<Int>,
```
Mutable타입을 사용해주셨네요.
하지만 Mutable타입을 사용하게 되면, 외부에서 데이터를 변경하게 되면 의도치 않게 내부의 데이터가 변경되어 무결성이 깨지게 돼요.
외부에서 데이터를 변경할 수 없도록, List타입을 활용해보면 어떨까요?

Reply:
외부에서 넘겨준 리스트가 Mutable 일 경우에는 얕은 복사가된것을 받아서 외부에서 리스트를 변경하면 의도치 않게 내부에서도 데이터가 변경될수 있으니 리스트로 받은후 변경이 필요하다면 toMutableList로 변경해서 사용하라는 말씀이신가요?

Reply:
네 맞습니다.
정확히는 "방어적 복사"를 활용하는 것입니다.
"방어적 복사"에 대해서 학습을 해보시고, 코드를 개선해보면 어떨까요?
특히 생성자의 타입은, 언급한바와 같이 Mutable이 아닌 Immutable 타입을 사용하는 것을 권장합니다.